File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool vowel(char c)
    {
        return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' ||
              c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U' );
    }
    bool halvesAreAlike(string s) {
        int i,n=s.size(),cnt=0;
        for(i=0;i<n/2;i++)
        {
            if(vowel(s[i]))
                 cnt++;
        }
        for(i=n/2;i<n;i++)
        {
            if(vowel(s[i]))
                 cnt--;
        }
        return cnt==0;
    }
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution(object):
    def halvesAreAlike(self, s):
        vowels = ['a','e','i','o','u','A','E','I','O','U']
        cnt= 0
        n=len(s)
        for i in range(n):
            if(i>=(n/2)):
                 if s[i] in vowels:
                        cnt-=1
            else:
                if s[i] in vowels:
                        cnt+=1
        return cnt==0
        